db/db.go

```diff
@@ -53,28 +52,33 @@ func (db *DB) Close() error {
 // RowResults.
 func (db *DB) Query(query string) (RowResults, error) {
 	if !strings.HasPrefix(strings.ToUpper(query), "SELECT ") {
-		log.Warn("Query \"%s\" may modify the database", query)
+		log.Warnf("Query \"%s\" may modify the database", query)
 	}
 	rows, err := db.dbConn.Query(query)
 	if err != nil {
-		log.Error("failed to execute SQLite query", err.Error())
+		log.Errorf("failed to execute SQLite query: %s", err.Error())
 		return nil, err
 	}
-	defer rows.Close()
+	defer func() {
+		err = rows.Close()
+		if err != nil {
+			log.Errorf("failed to close rows: %s", err.Error())
+		}
+	}()
 
 	results := make(RowResults, 0)
 
 	columns, _ := rows.Columns()
 	rawResult := make([][]byte, len(columns))
 	dest := make([]interface{}, len(columns))
-	for i, _ := range rawResult {
+	for i := range rawResult {
```

Comment:
Yes, it only works for indexes. For elements of the array, a `for _, result := range ...` notation is needed. This is something that `gometalinter` complained about.